components/organisms/Listener/sketch.jsx

```diff
@@ -0,0 +1,131 @@
+"use client";
+import { useEffect, useRef } from "react";
+import P5 from "p5";
+import PropTypes from "prop-types";
+import { useAladin } from "@/contexts/Aladin";
+import { automaticWalk, shiftStarTint } from "./utilities";
+import parameters from "./parameters";
+import PixelSynth from "./PixelSynth";
+
+const Sketch = ({ pixelColor }) => {
+  const sketchRef = useRef(null);
+  const p5Instance = useRef(null);
+  const currentColor = useRef(pixelColor);
+  const { aladin } = useAladin();
+  const emptyCircleRef = useRef(null);
+  const starRef = useRef(null);
+  const synthRef = useRef(null);
+
+  // Initialize synth
+  useEffect(() => {
+    synthRef.current = new PixelSynth();
+  }, []);
+
+  // Update the current color when pixelColor changes
+  useEffect(() => {
+    currentColor.current = pixelColor;
+    // Update synth frequency when color changes and mouse is pressed
+    if (synthRef.current && parameters.mouseIsPressed) {
+      synthRef.current.playNote(pixelColor);
+    }
+  }, [pixelColor]);
+
+  // Create the sketch only once
+  useEffect(() => {
```

Comment:
you might be doing this for other reasons, but if you want to wait on aladin to load and then do something exactly once, there's an `onLoaded` callback that you can pass to the `useAladin` hook which will pass the instance of `aladin` and it can be used for setup-y stuff.

`components/molecules/ExplorerControls/Orientation/index.tsx` has an example of it working to set up a widget with some data from the aladin instance on load